src/wp-admin/includes/class-plugin-upgrader.php

```diff
@@ -317,6 +317,8 @@ public function bulk_upgrade( $plugins, $args = array() ) {
 		$this->update_count   = count( $plugins );
 		$this->update_current = 0;
 		foreach ( $plugins as $plugin ) {
+			unset( $this->result );
```

Comment:
What does this do to line 350, now 352 below?

Reply:
`$this->result` is set by `this->run()` at line 339-350 (or is at least populated after? I could be missing some specific nuance here), but with the current behavior (before without these changes), on subsequent iterations though the loop, `$this->result` holds the output of the previous update, and if the secondary update fails, $this->result is still the output of update 1, so then `$results[ <plugin2> ]` ends up with faulty information.

To see this behavior, you can add print_r($results) after the whole loop and try the following:
```
$ wp plugin install akismet --version=4.0 --force
$ wp plugin install wordpress-importer --version=0.5 --force
$ chmod -w wp-content/plugins/wordpress-importer
$ wp plugin update --all
# > See "Success: Updated 2 of 2 plugins."
```

Reply:
There could be a better place/way to be setting or resetting `$this->result` that I should be using though? 🤷🏻